Reemit.Gui/Views/Controls/Icons/Icon.axaml.cs

```diff
@@ -49,8 +78,22 @@ private void UpdateIcon()
             IconKind.Method => new MethodIcon(),
             _ => throw new ArgumentOutOfRangeException(nameof(Kind), Kind, "Unrecognized icon kind")
         });
-            
+
+        AdjustIconLuminosity(icon);
+
         Children.Clear();
         Children.Add(icon);
     }
+
+    private void AdjustIconLuminosity(Control icon)
+    {
+        // This looks hack-ish, but works for all the VS icons we currently have.
+        if (icon.GetLogicalChildren().FirstOrDefault()?.GetLogicalChildren().FirstOrDefault() is Rectangle iconRectangle)
```

Comment:
Somewhat of a nitpick, but should this be rewritten to be more resilient against changes in the XAML? Preferrably, an approach that doesn't expect children at specific offsets. Not sure what the right level of genericism is here, but perhaps we could search for every `SolidColorBrush` in the control and adjust luminosity. I suppose it depends on whether we expect to ever have a control that contains a `SolidColorBrush` we don't want to transform.

Also, a bit more out there and quite out of scope for this PR, but: I wonder if we could find a way to move some of this to the XAML. Maybe a markup extension or something. However, probably not worth doing if we don't expect to transform luminosity outside of this.

Reply:
Good idea. Rewrote this to work regardless of the layout of the control tree.